internal/controller/streams.go

```diff
@@ -505,7 +509,30 @@ func ConnectStreams(ctx context.Context, logger *zap.SugaredLogger, envMap Envir
 				logger:             logger,
 				bufferedGrpcSyncer: bufferedGrpcSyncer,
 			}
-
+			ciliumFlowCollector, err := newCiliumFlowCollector(ctx, logger, envMap.CiliumNamespace)
```

Comment:
This is creating a Cilium flow collector just to check whether this succeeds?
This creates a gRPC client and doesn't close it, which basically may leak connections and goroutines.
We shouldn't have to do that.